unit_tests/test_exceptions.py

```diff
@@ -62,6 +62,37 @@ def test_hit_w_content_as_str(self):
         self.assertEqual(exception.message, 'Not Found')
         self.assertEqual(list(exception.errors), [])
 
+    def test_hit_w_content_as_unicode(self):
+        from google.cloud.exceptions import NotFound
+        _ERROR_MESSAGE = u'That\u2019s not found.'
+        _EXPECTED = (b'404 That\xe2\x80\x99s not found.',
+                     '404 %s' % (_ERROR_MESSAGE,))
+
+        response = _Response(404)
+        content = u'{"error": {"message": "%s" }}' % (_ERROR_MESSAGE,)
+
+        exception = self._callFUT(response, content)
+        self.assertIn(str(exception), _EXPECTED)
```

Comment:
Let's bow with a pragma:

```python
expected = u'404 ' + _ERROR_MESSAGE
if six.PY2:
    self.assertEqual(str(exception), _to_bytes(expected, encoding='utf-8'))
else:  # pragma: NO COVER
    self.assertEqual(str(exception), expected)
```

You only need the pragma on the branch which isn't traversed in Python 2.

----

ASIDE: This is a test, so you don't need CAPITAL varnames or underscores in the var names. You own the local scope.